cmake/dependencies.cmake

```diff
@@ -77,7 +77,7 @@ find_package(tbb)
 ##########################
 #         boost          #
 ##########################
-find_package(Boost 1.58.0 REQUIRED
+find_package(Boost 1.65.0 REQUIRED
```

Comment:
May you example why 1.65 (e.g not 1.59)?

Reply:
I can't find the exact line now, but it's related to pedantic warning in boost multiprecision: some macro was terminated with `;`when it is not required. This issue was fixed in version 1.65.